cpp/include/raft/linalg/detail/eig.cuh

```diff
@@ -52,7 +52,7 @@ void eigDC_legacy(raft::device_resources const& handle,
   rmm::device_uvector<math_t> d_work(lwork, stream);
   rmm::device_scalar<int> d_dev_info(stream);
 
-  raft::matrix::copy(in, eig_vectors, n_rows, n_cols, stream);
+  raft::copy_async(eig_vectors, in, n_rows * n_cols, stream);
```

Comment:
This is kind of a "quick fix" which is, in turn, coupling out detail APIs across namespaces, which is actually creating more tech debt. I'd prefer if instead of doing this, we would use the new matrix public APIs so that we can move closer to removing the deprecated APIs altogether.

If the goal here is really just to silence the deprecation warnings, there's other ways we can do that.

Reply:
thanks for reviewing this!  This is still a draft here (and there is still quite a few other things that need changed/fixed before we can merge this) - but the last couple commits remove most of the matrix::detail namespace hacks I had in here.